hypertools/plot/__init__.py

```diff
@@ -2,9 +2,7 @@
 import warnings
 import matplotlib as mpl
 
-try:
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore")
-        mpl.use('TkAgg')
-except:
-    warnings.warn('Could not switch backend to TkAgg.  This may impact performance of the plotting functions.')
+from .backend import _init_backend
+
+
+_init_backend()
```

Comment:
I haven't traced through the full _init_backend() pipeline-- it's important that this works both in notebooks *and* in "classic" python sessions.  all of these scenarios should result in working hypertools functions:
- jupyter notebook
- jupyter lab
- colaboratory
- ipython session
- python session (e.g. run from terminal)
- jupyter/ipython notebook emulators (e.g. pycharm, atom, vscode)
- when hypertools is imported for non-plotting functionality as part of other packages (e.g. timecorr, supereeg, etc.) in various settings

Reply:
The `_init_backend` function decides whether or not `hypertools` was imported into a notebook based on two checks:
1. The `get_ipython()` function exists in the global namespace
   - True for Jupyter notebooks, JupyterLab, and IPython shells, False otherwise
2. If yes, the config for the current IPython interactive shell instance stores a reference to an `IPKernelApp` instance
   - True for Jupyter notebooks and JupyterLab, False for IPython shells

If both checks pass, the plotting backend is initialized for a notebook environment
##

- **jupyter notebook**
- **jupyter lab**

  - These will pass both checks (JupyterLab's notebook-like editor is actually just a wrapper around the regular jupyter notebook app)

- **collaboratory**

  - Unfortunately, plotting backends and animation work differently enough in Colab from how they do in Jupyter that it's not possible to support interactive/animated plots in Colab notebooks at this point.  Most `matplotlib`-based libraries don't seem to deal with this issue (and we didn't support this previously, either), so I think this is acceptable for now.  That said, `plotly` is entirely immune to this issue since they use their own JS library for nearly everything, so if we support `plotly` as a plotting backend in the future, that can be the fallback for Colab.

- **ipython session**

  - This will pass the first check but fail the second (and was the impetus behind adding the second check)

- **python session (e.g. run from terminal)**

  - This will fail both checks (which is what we want)

- **jupyter/ipython notebook emulators (e.g. pycharm, atom, vscode)**

  - This is an interesting situation that I hadn't thought of.  Notebooks run through IDEs/editors _should_ pass both checks -- I know PyCharm [runs a proper notebook server](https://www.jetbrains.com/help/pycharm/jupyter-notebook-support.html#tool-windows) for its notebook interface, and it looks like VSCode [requires you start one separately, first](https://code.visualstudio.com/docs/python/jupyter-support#_setting-up-your-environment).  As for Atom, there are [a number](https://atom.io/packages/search?utf8=%E2%9C%93&q=keyword:jupyter) of 3rd-party packages that claim to do this and the two most popular ones both seem to run IPython kernels as well<sup>[1](https://nteract.io/kernels),[2](https://github.com/jupyter/atom-notebook/blob/6e1b766ca232d1149053f94f8ec3ce84b906ab1d/lib/notebook-editor.js#L12-L17)</sup>.

- **when hypertools is imported for non-plotting functionality as part of other packages (e.g. timecorr, supereeg, etc.) in various settings**
  - The backend is set for the correct environment no matter how `hypertools` is imported, since the `plot` function is imported into the top-level module namespace (so `plot/__init__.py` gets run):
      ```
      In [1]: import hypertools as hyp
      
      In [2]: plot1 = hyp.plot
      
      In [3]: from hypertools.plot.plot import plot as plot2
      
      In [4]: plot1 is plot2
      Out[4]: True
      ```
      This behavior was the source of #242 and should ideally be refactored out, though it would be at the cost of being able to call `hypertools.plot` directly (barring some hack I can't think of at the moment).